libkbfs/folder_branch_ops.go

```diff
@@ -3787,10 +3741,85 @@ func (fbo *folderBranchOps) syncAllLocked(
 			fbo.dirOps = nil
 		}
 	}()
-	for _, op := range fbo.dirOps {
-		md.AddOp(op.dirOp)
+
+	newBlocks := make(map[BlockPointer]bool)
+	fileBlocks := make(fileBlockMap)
+	parentsToAddChainsFor := make(map[BlockPointer]bool)
+	for _, dop := range fbo.dirOps {
+		// Copy the op before modifying it, in case there's an error
+		// and we have to retry with the original ops.  TODO: make a
+		// better way of copying a single op.
+		ops := make(opsList, 1)
+		err := kbfscodec.Update(
```

Comment:
If anyone prefers, I can add a `DeepCopy` method to the `op` interface, to make this cleaner (and probably more performant).